src/transformers/training_args.py

```diff
@@ -2498,6 +2498,11 @@ def to_dict(self):
             # Handle the accelerator_config if passed
             if is_accelerate_available() and isinstance(v, AcceleratorConfig):
                 d[k] = v.to_dict()
+            # Handle the BitsAndBytes quantization_config if passed
+            if k == "model_init_kwargs" and isinstance(v, dict) and "quantization_config" in v:
+                quantization_config = v.get("quantization_config")
+                if not isinstance(quantization_config, dict):
+                    d[k]["quantization_config"] = quantization_config.to_dict()
```

Comment:
This applies to any quantization configuration that’s compatible with training. Let’s update the comment to:
```suggestion
            # Handle the quantization_config if passed
            if k == "model_init_kwargs" and isinstance(v, dict) and "quantization_config" in v:
                quantization_config = v.get("quantization_config")
                if not isinstance(quantization_config, dict):
                    d[k]["quantization_config"] = quantization_config.to_dict()
```

Reply:
Thanks for the prompt review. I've included your suggestion.